### PR TITLE
add notes on Godep

### PR DESCRIPTION
**Pull request by natebrennand**
Example links will work after https://github.com/Clever/shorty/pull/10 is merged

### CODE REVIEW

File: golang/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Golang versions
 
-All golang services (with some exceptions) at Clever should be on go-1.5.
+All golang services (with some exceptions, like Heka) at Clever should be on go-1.5.
```
Comment (peggyl) [inline golang/README.md]: Reasonable to get a list of current exceptions, or what informal criteria are used for making exceptions?

File: golang/godep.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+If they are not included, please add them for consistency.
+
+```make
+SHELL := /bin/bash
+# use the Go toolchain to get a list of every Go package, we filter out packages in the vendor directory
+PKGS := $(shell go list ./... | grep -v /vendor)
+# thrift services should also filter out "gen-go"
+# PKGS := $(shell go list ./... | grep -v /vendor | grep -v /gen-go)
+
+# declaration of godep as well as assuring that the most recent version is in use
```
Comment (peggyl) [inline golang/godep.md]: Nit: s/assuring/ensuring/

File: golang/godep.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+│           ├── gopkg.in
+│               ├── mgo (v2)
+```
+
+### Existing Packages
+
+First ensure that you have your desired version of the package checked out in your `$GOPATH`.
+In our structure, this means that we now have `v2` of `github.com/Clever/worker` checked out.
+
+When to change the version of an existing package, you will need to use the godep tool manually without a Makefile target.
```
Comment (peggyl) [inline golang/godep.md]: Nit: s/to change/changing

File: golang/README.md (hunk 1 above)
Comment (natebrennand) [inline golang/README.md]: As far as @cgclever and I know the only exception is Heka. I'm going to note that all "services written by Clever" should be Go1.5 and acknowledge that 3rd party exceptions exist

**Pull request merged**